graphql/query_resolver.go

```diff
@@ -982,44 +982,79 @@ func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions)
 		requesters = evergreen.SystemVersionRequesterTypes
 	}
 
+	maxOrderOpt := utility.FromIntPtr(options.MaxOrder)
+	minOrderOpt := utility.FromIntPtr(options.MinOrder)
+
 	opts := model.WaterfallOptions{
 		Limit:      limit,
 		Requesters: requesters,
+		MaxOrder:   maxOrderOpt,
+		MinOrder:   minOrderOpt,
+	}
+
+	activeVersions, err := model.GetActiveWaterfallVersions(ctx, projectId, opts)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting active waterfall versions: %s", err.Error()))
+	}
+
+	// Since GetAllWaterfallVersions uses an inclusive order range ($gte instead of $gt), add 1 to our minimum range
+	minVersionOrder := minOrderOpt + 1
+	if minOrderOpt == 0 {
+		minVersionOrder = activeVersions[len(activeVersions)-1].RevisionOrderNumber
+	}
+
+	// Same as above, but subtract for max order
+	maxVersionOrder := maxOrderOpt - 1
+	if maxOrderOpt == 0 {
+		maxVersionOrder = activeVersions[0].RevisionOrderNumber
```

Comment:
is it possible to just set 
```
minVersionOrder = activeVersions[len(activeVersions)-1].RevisionOrderNumber
maxVersionOrder = activeVersions[0].RevisionOrderNumber
```
without the other if statements here? (I just did very basic example on my notebook so I might have missed something)

Reply:
This situation occurs when there is an inactive version between `maxOrder` (i.e. the last version on the previous page) and the zeroth active version. That version must also be retrieved to display as an inactive commit. The `page_forward.graphql` test catches this; I ran a failing patch with your changes [here](https://spruce.mongodb.com/task/evergreen_ubuntu2204_test_service_graphql_patch_2ee4bc12ab49f8417a221f3606f21b184dd0c912_66e205b32526dc0007ae3487_24_09_11_21_04_07/logs?execution=0).

Sorry for the lack of clarity, I will add comments!